### ISSUE TITLE
no such time zone: 'ATLANTIC/AZORES': U_ILLEGAL_ARGUMENT_ERROR" 

### ISSUE THREAD

**Issue by KmeCnin** [rlanvin/php-rrule#103]
Version used: `v2.3.0`

When calling `RRule::humanReadable` with `$timezone === 'ATLANTIC/AZORES'` returned formatter is `null` resulting in the following exception beeing thrown:
```
"IntlDateFormatter::create() failed. Error Code: 1 "datefmt_create: no such time zone: 'ATLANTIC/AZORES': U_ILLEGAL_ARGUMENT_ERROR" (this should not happen, please open a bug report!)".
```
Called in a context of a rendering in a Twig template:
```twig
<span title="{{ object.rRule.rfcString }}">{{ object.rRule.humanReadable|capitalize }}</span>
```

**Comment by rlanvin** [comment]
Hello, so as far as I know, timezone identifiers (TZID) are case sensitive and have a specific capitalization. The valid identifier is `Atlantic/Azores`.

Now what is interesting, is that it accepted this during the instantiation of the object. The reason this exception says "it shouldn't happen", is because the `DateTimeZone` object is created when the `RRule` is created, so if the timezone wouldn't be valid, it should fail in the constructor, not at the `humanReadable()` stage.

And indeed, this works fine:
```php
$tz = new DateTimeZone('ATLANTIC/AZORES');
var_dump($tz);
```
and will return a valid timezone:
```
class DateTimeZone#10 (2) {
  public $timezone_type =>
  int(3)
  public $timezone =>
  string(15) "ATLANTIC/AZORES"
}
```

But when this timezone is added into `IntlDateFormatter` (used for `humanReadable`), it fails:

```php
$frmt = \IntlDateFormatter::create(
    'en',
    \IntlDateFormatter::SHORT,
    \IntlDateFormatter::SHORT,
    new DateTimeZone('ATLANTIC/AZORES')
);
var_dump($frmt); // NULL
```

In other words, it looks like you discovered an inconsistency in PHP Intl library regarding timezone identifiers: the `DateTimeZone` object is case insensitive but the `IntlDateFormatter` object is case sensitive. I don't know why.

I'm not sure this is fixable in this library, since the problem is in the underlying Intl library (or maybe even in the underlying ICU library). In any case, the fix would only make the exception happen earlier - you should use `Atlantic/Azores`.

**Comment by KmeCnin** [comment]
Thanks a lot for this super fast and cristal clear answer!

And thanks for your work on this super useful library :)

**Comment by rlanvin** [comment]
Hello @jdbargy, what do you mean "the uppercase of TimeZone is done by `createFromRfcString`."? As far as I know `createFromRfcStringDo` doesn't make the timezone uppercase. Do you have a code example so I can understand the problem and try to reproduce it locally?

**Comment by jdbargy** [comment]
Hello,
For example when we have an rfc String like this ```
DTSTART;TZID=Europe/Paris:20200929T000000
RRULE:FREQ=DAILY;BYSECOND=0;BYMINUTE=0;BYHOUR=9
``` and we want to create a RRule from it, the createFromRfcString method of RRule will change it into ```
DTSTART;TZID=EUROPE/PARIS:20200929T000000
RRULE:FREQ=DAILY;BYSECOND=0;BYMINUTE=0;BYHOUR=9
```

**Comment by rlanvin** [comment]
`createFromRfcString` indeed had a bug and forced uppercase. I published v2.3.2 that solves this. Thanks for the report!

**Comment by jdbargy** [comment]
Thanks a lot for this fix and for your reactivity.
